src/main/main.ts

```diff
@@ -120,7 +120,7 @@ export type Client = {
   verifyMfaEmailRegistration: (params: VerifyMfaEmailRegistrationParams) => Promise<void>
   verifyMfaPasswordless: (params: VerifyMfaPasswordlessParams) => Promise<AuthResult>
   verifyMfaPhoneNumberRegistration: (params: VerifyMfaPhoneNumberRegistrationParams) => Promise<void>
-  verifyPasswordless: (params: VerifyPasswordlessParams, options?: AuthOptions) => Promise<void>
+  verifyPasswordless: (params: VerifyPasswordlessParams, options?: AuthOptions) => Promise<AuthResult | void>
```

Comment:
Perhaps just return `Promise<AuthResult>` ? If a redirection is used anyway, even `Promise<void>` isn't very useful 😅

Reply:
Non on ne peut pas faire ça, on ets obligé de type le retour de l'embranchement qui fait la redirection.

`location.assign` retourne un type `void`, donc soit un se retrouve avec un type de retour de `verifyPasswordless` qui est `Promise<AuthResult> | void`, soit comme dans le cas présent, on fait en sorte d'avoir une `Promise` en retour dans tous les cas de figure pour faciliter la gestion du retour de la méthode.